daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -88,7 +89,11 @@ func init() {
 // the EC2 Instance Metadata Service.
 func New(ctx logger.Context) (logger.Logger, error) {
 	logGroupName := ctx.Config[logGroupKey]
-	logStreamName := ctx.ContainerID
+	logStreamName, err := loggerutils.ParseLogTag(ctx, loggerutils.DefaultTemplate)
```

Comment:
I don't believe this to should trigger an error - Specifying `awslogs-stream` with tag will have no error, as `awslogs-stream` is "king" and overwrites the tag specified.
 
This allows you to update the driver to use the tag specified when no logstream have been specified, but not see any change in currently running containers that have the `--log-opt awslogs-stream="someApp"` (keeps this driver backwards compatible)